WordPress/gutenberg
# Add Reusable block save button, snackbar on save and Welcome Guide

Fixes https://github.com/WordPress/gutenberg/issues/32020

## Description
Added a Save Button for Reusable Blocks in Blocks Toolbar.  If there are no changes, the save button is disabled. When a change is made in that particular reusable block. One can save the changes from the block toolbar. Once saved, the user is notified with a snackbar about the save. The snackbar's 'Learn More' opens the Reusable Block Welcome Guide which also has a link to external documentation.

Since the image (gif) for the welcome guide is not ready yet. I have kept a plain background in its place.
I will add the image whenever it's made available.

## How has this been tested?
1. Select a Reusable block
2. Make Changes
3. Save the Reusable Block from Toolbar
4. A snackbar should appear at the bottom left 
5. Click on Snackbar's Learn More. This should open the Welcome guide.


## Screenshots 

https://user-images.githubusercontent.com/69596988/120833132-117f3580-c57f-11eb-8a39-1626f89f28e5.mov




## Types of changes





## Checklist:
